### PR TITLE
Fixed Event StartingRecallEventArgs's IsAllowed issue.

### PR DESCRIPTION
**Pull request by XKaguya**
StartingRecallEventArgs's IsAllowed won't stop the event. Patch at RagdollAbilityBase could solve this.

### CODE REVIEW

**Comment by XKaguya** [comment]
The part of `                // base.CurRagdoll
                new(OpCodes.Ldarg_1),
                new(OpCodes.Callvirt, PropertyGetter(typeof(RagdollAbilityBase<Scp049Role>), nameof(RagdollAbilityBase<Scp049Role>.CurRagdoll))),` currently will lead into crash , dont merge before if i can fix this bug.Another option is to give up acquiring Target players, although I know this is not realistic, lol

**Comment by louis1706** [comment]
this will also be call for Scp0492

**Comment by XKaguya** [comment]
> this will also be call for Scp0492 'RagdollAbilityBase'

yes , that is what the arg used for.Actually it will be call by Scp3114, but to doing limits on the Player , then it can be avoided.

**Comment by XKaguya** [comment]
The problem about previous patch's ```IsAllowed``` is not vailed is because NW using non-zero as error code.Meanwhile ```Transpiler``` can't modify the return value.
So the patch must use ```Prefix``` instead of ```Transpiler ```.
Things are all done now.This event can get ```Target```, ```Player```, ```Ragdoll``` and ```IsAllowed```.

**Comment by louis1706** [comment]
> IsAllowed

Transpiler can modify return value ???

**Comment by XKaguya** [comment]
> > IsAllowed
> 
> Transpiler can modify return value ???

uh , letme try that lol , forget its just stack

**Comment by XKaguya** [comment]
> > IsAllowed
> 
> Transpiler can modify return value ???

Alright. now i've rewritten as Transpiler.And tested.

**Comment by Misaka-ZeroTwo** [comment]
Please target dev branch.

File: Exiled.Events/EventArgs/Scp049/StartingRecallEventArgs.cs [hunk 1]
```diff
@@ -1,66 +1,66 @@
...
+        /// <param name="scp049">
+        ///     <inheritdoc cref="Player" />
+        /// </param>
+        /// <param name="ragdoll">
+        ///     <inheritdoc cref="Ragdoll" />
+        /// </param>
+        /// <param name="isAllowed">
+        ///     <inheritdoc cref="IsAllowed" />
+        /// </param>
+        public StartingRecallEventArgsMod(Player player, Ragdoll ragdoll, bool isAllowed = true)
```
Comment (Misaka-ZeroTwo) [inline Exiled.Events/EventArgs/Scp049/StartingRecallEventArgs.cs:34]:
```suggestion
        public StartingRecallEventArgs(Player player, Ragdoll ragdoll, bool isAllowed = true)
```

**Review by Misaka-ZeroTwo — CHANGES_REQUESTED** [review]

**Comment by XKaguya** [comment]
> Please target dev branch.

OK I'll do it after i get up

**Comment by Misaka-ZeroTwo** [comment]
You will need to create a separate PR to prevent conflicts.

**Pull request closed without merging**